apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -58,14 +58,25 @@
     private final SpanContext context = new SpanContext();
     @Nullable
     private Throwable stacktrace;
+    @Nullable
+    private AbstractSpan<?> parent;
+    @Nullable
+    private Transaction transaction;
 
     public Span(ElasticApmTracer tracer) {
         super(tracer);
     }
 
     public <T> Span start(TraceContext.ChildContextCreator<T> childContextCreator, T parentContext, long epochMicros, boolean dropped) {
-        onStart();
         childContextCreator.asChildOf(traceContext, parentContext);
+        if (parentContext instanceof Transaction) {
```

Comment:
Since this is where you make the parent and transaction references, I think this is the best place to update reference counters. This way you won't miss.